bin/tx-prover/src/proxy/mod.rs

```diff
@@ -47,22 +72,168 @@ impl LoadBalancer {
+    /// If the worker is not in the list, it won't be added.
     pub async fn add_available_worker(&self, worker: Backend) {
-        let mut available_workers = self.available_workers.write().await;
-        assert!(!available_workers.contains(&worker), "Worker already available");
-        info!("Worker {} is now available", worker.addr);
-        available_workers.push(worker);
+        let mut available_workers = self.current_workers.write().await;
+        if let Some(w) = available_workers.iter_mut().find(|w| w.worker == worker) {
+            w.is_available = true;
+        }
+    }
+
+    /// Updates the list of available workers based on the given action ("add" or "remove").
+    ///
+    /// # Behavior
+    ///
+    /// ## Add Action
+    /// - If the worker exists in the current workers list, do nothing.
+    /// - Otherwise, add it and mark it as available.
+    ///
+    /// ## Remove Action
+    /// - If the worker exists in the current workers list, remove it.
+    /// - Otherwise, do nothing.
+    ///
+    /// Finally, updates the configuration file with the new list of workers.
+    ///
+    /// # Errors
+    /// - If the worker cannot be created.
+    /// - If the configuration cannot be loaded.
+    /// - If the configuration cannot be saved.
+    pub async fn update_workers(
+        &self,
+        update_workers: UpdateWorkers,
+    ) -> std::result::Result<(), String> {
+        let mut current_workers = self.current_workers.write().await;
+
+        info!("Current workers: {:?}", current_workers);
+
+        let workers: Vec<Worker> = update_workers
+            .workers
+            .iter()
+            .map(|worker| Backend::new(worker))
+            .collect::<Result<Vec<Backend>, _>>()
+            .map_err(|err| format!("Failed to create backend: {}", err))?
+            .into_iter()
+            .map(Worker::new)
+            .collect();
+
+        match update_workers.action {
+            Action::Add => {
+                for worker in workers {
+                    if !current_workers.iter().any(|w| w.worker == worker.worker) {
+                        current_workers.push(worker);
+                    }
+                }
+            },
+            Action::Remove => {
+                for worker in workers {
+                    current_workers.retain(|w| w.worker != worker.worker);
+                }
+            },
+        }
+
+        let mut configuration = ProxyConfig::load_config_from_file()
+            .map_err(|err| format!("Failed to load configuration: {}", err))?;
+
+        let new_list_of_workers = current_workers
+            .iter()
+            .map(|worker| {
+                worker
+                    .worker
+                    .as_inet()
+                    .ok_or_else(|| "Failed to get worker address".to_string())
+                    .map(|worker_addr| {
+                        WorkerConfig::new(&worker_addr.ip().to_string(), worker_addr.port())
+                    })
```

Comment:
Could this be replaced by something like `Worker::address()` method? (i.e., move this logic into the `Worker` struct).